browser/BUILD.gn

```diff
@@ -11,8 +11,18 @@ source_set("browser_process") {
     "bookmarks/brave_bookmark_client.h",
     "brave_browser_main_extra_parts.cc",
     "brave_browser_main_extra_parts.h",
+    "brave_browser_main_parts.cc",
+    "brave_browser_main_parts.h",
+    "brave_browser_main_parts_android.cc",
```

Comment:
we don't need all of these _os files because none of them implement `PreShutdown`

Reply:
ChromeBrowserMainParts<OS> inherit from ChromeBrowserMainParts (except Linux and Mac which both inherit from Posix). If I just subclass ChromeBrowserMainParts I would still have to change ChromeBrowserMainParts<Win,Android,Posix> to inherit from the new subclass (see instantiation in https://cs.chromium.org/chromium/src/chrome/browser/chrome_content_browser_client.cc?l=1150). 
Also, while none of them override PreShutdown now I am not sure we can depend on it staying this way or, if they do override, that they'd call the base class implementation.